awx/main/models/receptor.py

```diff
@@ -0,0 +1,28 @@
+from django.db import models
+
+from awx.main.models.base import BaseModel
+from awx.main.models.ha import Instance
+
+
+class ReceptorNode(BaseModel):
+    NODE_TYPE_CHOICES = [
+        ('control', "Control plane node"),
+        ('hybrid', "Controller and execution"),
+        ('execution', "Execution plane node"),
+        ('hop', "Hop node"),
+    ]
+    NODE_STATE_CHOICES = [
+        ('healthy', "Healthy"),
+        ('unhealthy', "Unhealthy"),
+        ('disabled', "Disabled"),
```

Comment:
The disabled state would be duplicated with the instance model. I can maybe see through to the intention with the unhealthy state being a receptor construct, but this is also duplicated with the `last_seen` attribute on the instance model. While it's true that doesn't record the `last_seen` for hop nodes, hop nodes can't be disabled... so this starts to look like a little bit of a data model mess.

Reply:
There is definitely something to what you say, but this is information that @kialam wants to include in the visualizer.  Do we want the API list view for `ReceptorNode`s to prefetch the `Instance`s, or is it ultimately acceptable for this to be denormalized in the database the way I have it here.